src/content/en/2021/resource-hints.md

```diff
@@ -1,20 +1,459 @@
+  description="Bar chart showing the adoption of rel=\"dns-prefetch\" segmented by CrUX rank. rel=\"dns-prefetch\" has a 29% for the top 1,000 sites on mobile and 28% on desktop. rel=\"dns-prefetch\" has a 25% for the top 10,000 sites on mobile and 25% on desktop. rel=\"dns-prefetch\" has a 23% for the top 100,000 sites on mobile and 23% on desktop. rel=\"dns-prefetch\" has a 27% for the top 1 million sites on mobile and 27% on desktop. rel=\"dns-prefetch\" has a 36% for the top 10 million sites on mobile and 36% on desktop.",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vTgxCYoH2vDbhTky1qQKEeV034kReHF8JYYq0aDyoo6LG22XL6Tar3dNPsqU1_zyvR6vuFEdMGoF1oP/pubchart?oid=1340595902&format=interactive",
+  sheets_gid="1880502987",
+  sql_file="hints_adoption_by_rank.sql"
+  )
+}}
+
+`dns-prefetch` is the only resource hint which exhibits a decrease in adoption when comparing the top 1,000 sites with the overall adoption.
+
+{{ figure_markup(
+  image="rel-preconnect-adoption-by-rank.png",
+  caption="Adoption of rel=\"preconnect\" segmented by CrUX rank.",
+  description="Bar chart showing the adoption of rel=\"preconnect\" segmented by CrUX rank. rel=\"preconnect\" has a 30% for the top 1,000 sites on mobile and 29% on desktop. rel=\"preconnect\" has a 25% for the top 10,000 sites on mobile and 25% on desktop. rel=\"preconnect\" has a 18% for the top 100,000 sites on mobile and 18% on desktop. rel=\"preconnect\" has a 14% for the top 1 million sites on mobile and 14% on desktop. rel=\"preconnect\" has a 13% for the top 10 million sites on mobile and 13% on desktop.",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vTgxCYoH2vDbhTky1qQKEeV034kReHF8JYYq0aDyoo6LG22XL6Tar3dNPsqU1_zyvR6vuFEdMGoF1oP/pubchart?oid=1818271680&format=interactive",
+  sheets_gid="1880502987",
+  sql_file="hints_adoption_by_rank.sql"
+  )
+}}
+
+To counter this decrease, the top 1,000 pages have an increased adoption for the `preconnect` hint taking advantage of the benefits covered earlier on in the chapter.
+
+### Usage
+
+Resource hints can be very effective if used correctly. By shifting the responsibility from the browser to the developer, it allows you to prioritise resources required for the critical rendering path and improve the load times & user experience.
+
+<figure markdown>
+| rank       | preload | prefetch | preconnect | prerender | dns-prefetch | modulepreload |
+|------------|---------|----------|------------|-----------|--------------|-------------- |
+| 1,000      |       3 |        2 |          4 |         0 |            4 |             1 |
+| 10,000     |       3 |        1 |          4 |         1 |            3 |             1 |
+| 100,000    |       2 |        2 |          3 |         1 |            3 |             1 |
+| 1,000,000  |       2 |        2 |          2 |         1 |            2 |             1 |
+| 10,000,000 |       2 |        2 |          1 |         1 |            2 |             1 |
```

Comment:
Since we have less than 10million sites we should avoid labelling that bucket as such.

```suggestion
| all |       2 |        2 |          1 |         1 |            2 |             1 |
```